skfda/preprocessing/dim_reduction/feature_extraction/_fda_feature_union.py

```diff
@@ -0,0 +1,115 @@
+            ignored.
+        n_jobs: int
+            Number of jobs to run in parallel.
+            ``None`` means 1 unless in a :obj:`joblib.parallel_backend`
+            context.
+            ``-1`` means using all processors.
+            The default value is None
+        transformer_weights: dict
+            Multiplicative weights for features per transformer.
+            Keys are transformer names, values the weights.
+            Raises ValueError if key not present in ``transformer_list``.
+        verbose: bool
+            If True, the time elapsed while fitting each transformer will be
+            printed as it is completed. By default the value is False
+        array_output: bool
+            indicates if the transformed data is requested to be a NumPy array
+            output. By default the value is False.
+
+    Examples:
+    Firstly we will import the Berkeley Growth Study data set
+    >>> from skfda.datasets import fetch_growth
+    >>> X,y = fetch_growth(return_X_y=True)
+
+    Then we need to import the transformers we want to use. In our case we
+    will use Minimum Redundancy Maximum Relevance.
+    Evaluation Transformer returns the original curve, and as it is helpful,
+    we will concatenate it to the already metioned transformer.
+    >>> from skfda.preprocessing.dim_reduction.feature_extraction import (
+    ...     FdaFeatureUnion,
+    ... )
+    >>> from skfda.preprocessing.dim_reduction.variable_selection import (
+    ...     MinimumRedundancyMaximumRelevance,
+    ... )
+    >>> from skfda.representation import EvaluationTransformer
+    >>> import numpy as np
+
+    Finally we apply fit and transform.
+    >>> union = FdaFeatureUnion(
+    ...     [
+    ...        ("mrmr", MinimumRedundancyMaximumRelevance()),
+    ...        ("eval", EvaluationTransformer()),
+    ...     ],
+    ...     array_output=True,
+    ... )
+    >>> np.around(union.fit_transform(X,y), decimals = 2)
+      array([[ 194.3,   81.3,   84.2, ...,  193.8,  194.3,  195.1],
+            [ 177.4,   76.2,   80.4, ...,  176.1,  177.4,  178.7],
+            [ 171.2,   76.8,   79.8, ...,  170.9,  171.2,  171.5],
+            ...,
+            [ 166.3,   68.6,   73.6, ...,  166. ,  166.3,  166.8],
+            [ 168.4,   79.9,   82.6, ...,  168.3,  168.4,  168.6],
+            [ 168.9,   76.1,   78.4, ...,  168.6,  168.9,  169.2]])
+    """
+
+    def __init__(
+        self,
+        transformer_list,
+        *,
+        n_jobs=None,
+        transformer_weights=None,
+        verbose=False,
+        array_output=False,
+    ) -> None:
+        self.array_output = array_output
+        super().__init__(
+            transformer_list,
+            n_jobs=n_jobs,
+            transformer_weights=transformer_weights,
+            verbose=verbose,
+        )
+
+    def _hstack(self, Xs) -> Union[DataFrame, ndarray]:
+
+        if self.array_output:
+            for i in Xs:
+                if isinstance(i, (FDataGrid, FDataBasis)):
```

Comment:
Why not the superclass instead?
```suggestion
                if isinstance(i, FData):
```